### PR TITLE
test: increase timeout for getting pod list for logs

### LINKED ISSUE #11320: CI: provide agent logs when cilium pods are in crash loopback
When agent's pod goes into crash loopback and therefore CI results into failure, it would be helpful to see why that is happening. Right now the output is not useful, e.g. the zip file doesn't collect the agent log. [Example dump](https://github.com/cilium/cilium/files/4576627/f3ee8175_K8sChaosTest_Connectivity_demo_application_Endpoint_can_still_connect_while_Cilium_is_not_running.1.zip) from such CI run where the pod went into this state when init.sh was failing due to compilation error. The issue does not happen when the CI is run locally.

### PR DESCRIPTION
**Pull request by nebril**
Also added explicit error check to this kubectl call

fixes #11320

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -863,13 +863,19 @@ func (kub *Kubectl) GetAllPods(ctx context.Context, options ...ExecOptions) ([]v
...
 		fmt.Sprintf("%s get pods --all-namespaces -o json", KubectlCmd),
-		ExecOptions{SkipLog: ops.SkipLog}).Unmarshal(&podsList)
+		ExecOptions{SkipLog: ops.SkipLog})
+
+	if !res.WasSuccessful() {
+		return nil, res.GetError()
+	}
+
+	err := res.Unmarshal(&podsList)
 	if err != nil {
```
Comment (errordeveloper) [inline test/helpers/kubectl.go:879]: so it used to lie here... I see!

**Review by errordeveloper — APPROVED** [review]

**Comment by nebril** [comment]
test-4.9

**Comment by nebril** [comment]
test-runtime

**Comment by nebril** [comment]
test-4.19

**Comment by nebril** [comment]
test-net-next

**Pull request merged**